topcoder-platform/community-app
# [$50] - Error on accessing /my-submissions is not handled gracefully #696

### What does this PR do?
It ensures error thrown when a user tries to access a challenge where he/she is not registered is handled gracefully for best user experience

### Description of Task to be completed?
- Validate properties gotten from submission phase date before returning props
- Add a new case to access denied page template to check if a user has not registered for a challenge when trying to access the route, then give the appropriate graceful error message

### How should this be manually tested?
1. Pull in latest changes from remote repository to local repository
2. Fire up terminal run `npm install`
3. Run `npm run dev` to start up application in development environment
4. Visit url supplied by the application usually **http://localhost:3000**, navigate to login page
5. login with `username: dan_developer` and `password: dantopcoder123` for normal user access
6. Visit **all challenges** page, then view the challenge specs for one you've not registered in.
7. You should then see the graceful error message

### Any background context you want to provide?
The JavaScript breaks due to a property `actualStartTime` being accessed on a not-validated `submissionPhaseDate` object.

### What are the relevant issues tracker number?
[$50] - Error on accessing /my-submissions is not handled gracefully #696

### Screenshots (if appropriate)
<img width="1439" alt="screen shot 2018-01-10 at 11 52 26 am" src="https://user-images.githubusercontent.com/20375577/34769039-ec45eb06-f5fc-11e7-9f17-5ae6c1093568.png">

### Questions:
If you have any question on this PR kindly comment below. Thanks
